library/azure_rm_roledefinition.py

```diff
@@ -0,0 +1,382 @@
+        self.assignable_scopes = None
+
+        self.results = dict(
+            changed=False,
+            id=None,
+        )
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        self._client = None
+
+        super(AzureRMRoleDefinition, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                    supports_check_mode=True,
+                                                    supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+
+        old_response = None
+        response = None
+        to_be_updated = False
+
+        # get management client
+        self._client = self.get_mgmt_svc_client(AuthorizationManagementClient,
+                                                base_url=self._cloud_environment.endpoints.resource_manager)
+
+        # build cope
+        self.scope = build_scope()
+
+        # get existing role definition
+        old_response = self.get_roledefinition()
+
+        if old_response:
+            self.results['id'] = old_response['id']
+
+        if self.state == 'present':
+            # check if the role definition exists
+            if not old_response:
+                self.log("Role definition doesn't exist in this scope")
+
+                to_be_updated = True
+                self.to_do = Actions.CreateOrUpdate
+
+            else:
+                # existing role definition, do update
+                self.log("Role definition already exists")
+                self.log('Result: {0}'.format(old_response))
+
+                # compare if role definition changed
+                if self.check_update(old_response):
+                    to_be_updated = True
+                    self.to_do = Actions.CreateOrUpdate
+
+        elif self.state == 'absent':
+            if old_response:
+                self.log("Delete role defintion")
+                self.results['changed'] = True
+
+                if self.check_mode:
+                    return self.results
+
+                self.delete_roledefinition()
+
+                self.log('role definition deleted')
+
+            else:
+                self.fail("role definition {0} not exists.".format(self.name))
+
+        if to_be_updated:
+            self.log('Need to Create/Update role definition')
+            self.results['changed'] = True
+
+            if self.check_mode:
+                return self.results
+
+            if self.to_do == Actions.CreateOrUpdate:
+                response = self.create_update_roledefinition()
+
+                self.results['id'] = response['id']
+
+        return self.results
+
+    # build scope
+    def build_scope(self):
+        subscription_scope = '/subscription/' + self.subscription_id
+        if self.scope is None:
```

Comment:
could be:
if self.scope is not None:
  return self.scope
else:
  return '/subscription/' + self.subscription_id